### PR TITLE
Update dynamic_layers.py and upscaling utilities.py

### PR DESCRIPTION
**Pull request by jbwhitco**
Changed line 209 to have the variable 'elements' formed based on just the basename of the UAVSAR file, such that element[1] is always the date string of the UAVSAR file.

### CODE REVIEW

**Comment by jbwhitco** [comment]
Hopefully this pull request includes the following two changes:
1) Minor bug correction on line 209 of dynamic layers.py.
2) Replace upscaling_utilities.py with a python-only (i.e., no xml) version.

File: soilscape_upscaling/upscaling_utilities.py [hunk 1]
```diff
@@ -5,15 +5,14 @@
 Dan Clewley & Jane Whitcomb
 
 General purpose utilities
-
-This file is licensed under the GPL v3 Licence. A copy of this
```
Comment (danclewley) [inline soilscape_upscaling/upscaling_utilities.py]: Need to keep licence text in header.

File: soilscape_upscaling/upscaling_utilities.py [hunk 2]
```diff
@@ -29,92 +28,101 @@ def get_gdal_format(file_name):
...
-
+    
     return gdalStr
 
-def colour_sm_image(inimage, outimage, max_value=0.5, band=1):
-    """
-    Colour image using RSGISLib XML interface
 
-    FIXME: This function should be tidied up possibly to use
+def colour_SM_image(inimage, colorimage, max_value=0.5, band=1):
```
Comment (danclewley) [inline soilscape_upscaling/upscaling_utilities.py]: Would be better to use the English spelling `colourimage` for consistency.

File: soilscape_upscaling/upscaling_utilities.py [hunk 3]
```diff
@@ -29,92 +28,101 @@ def get_gdal_format(file_name):
...
+            bandDefns.append(BandDefn(SMclass,color[i],1))
+            if i == 0: 
+                expression = '('+SMclass
+            else:
+                expression = expression+'+'+SMclass
+        expression = expression+')'
+        print('expression: ',expression)
+        imagecalc.bandMath(colorimage, expression, gdalformat, datatype, bandDefns)       
+        for i in range(11):
+            os.system('rm '+color[i])
```
Comment (danclewley) [inline soilscape_upscaling/upscaling_utilities.py]: Better to use `os.remove` here, should also work under Windows. Alternatively could use a temp directory (`tempfile.mkdtemp`) for the individual colour bands and remove the entire thing using `shutil.rmtree` which would also take care of any  .aux.xml ancillary files.

**Review by danclewley — CHANGES_REQUESTED** [review]
Thanks for the updates. I've made some minor changes it would be good to make before merging. As some stage I'll try and update so it doesn't require creating 11 new images before colouring.

**Comment by jbwhitco** [comment]
Corrected header text and variable names and streamlined removal of temporary files.

**Pull request merged**